adminpage/views.py

```diff
@@ -1,3 +1,272 @@
 from django.shortcuts import render
-
+from codex.baseerror import *
+from codex.baseview import APIView
+from django.contrib.auth.models import User
+from django.contrib.auth import authenticate, login, logout
+from wechat.models import Activity
+from wechat.models import Ticket
+from wechat.views import CustomWeChatView
+from WeChatTicket.settings import SITE_DOMAIN
+import time
+from datetime import datetime, timedelta
 # Create your views here.
+
+
+class Login(APIView):
+
+    def get(self):
+        if not self.request.user.is_authenticated():
+            raise ValidateError('User not logged in')
+        return 0
+
+    def post(self):
+        self.check_input('username', 'password')
+        username = self.input['username']
+        password = self.input['password']
+        user = authenticate(username=username, password=password)
+
+        if not username:
+            raise ValidateError('Username is empty！')
+        if not password:
+            raise ValidateError('Password is empty！')
+        if not User.objects.filter(username=username):
+            raise ValidateError('Username does not exist！')
+        if not user:
+            raise ValidateError('Wrong password！')
+
+        if user.is_active:
+            login(self.request, user)
+            return 0
+
+
+class Logout(APIView):
+
+    def post(self):
+        if self.request.user.is_authenticated():
+            logout(self.request)
+        else:
+            raise ValidateError('Logout failed！User has not logged in.')
+
+
+class ActivityList(APIView):
+
+    def get(self):
+        if not self.request.user.is_authenticated():
+            raise ValidateError('User not logged in！')
+
+        activityList = Activity.objects.all()
+        data = []
+        for item in activityList:
```

Comment:
Maybe you should use objects' filter to choose the item you want from `activityList` instead of using a loop to filter.